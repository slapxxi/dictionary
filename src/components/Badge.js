import { connect } from 'react-redux';
import glamorous from 'glamorous';
import { theme } from '../lib/constants';
import { nonViewedCount } from '../store/dictionary/selectors';

const Badge = glamorous.div(
  {
    boxSizing: 'border-box',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.badge,
    position: 'absolute',
    borderRadius: '50%',
    verticalAlign: 'middle',
    padding: 2,
    minWidth: 20,
    minHeight: 20,
    fontSize: 12,
    fontFamily: 'Arial, sans-serif',
    bottom: '100%',
    left: '100%',
    marginLeft: -15,
    marginBottom: -15,
    color: theme.text,
  },
  ({ children }) => ({
    display: children === 0 ? 'none' : 'flex',
  }),
);

const enhance = connect((state) => ({
  children: nonViewedCount(state),
}));

export default enhance(Badge);
