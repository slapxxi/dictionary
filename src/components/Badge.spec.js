import React from 'react';
import { render } from 'enzyme';
import { Badge } from './Badge';

it('renders', () => {
  const component = render(<Badge>10</Badge>);
  expect(component).toMatchSnapshot();
});
