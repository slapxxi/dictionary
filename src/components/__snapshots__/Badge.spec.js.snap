// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.css-0,
[data-css-0] {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background-color: tomato;
  position: absolute;
  border-radius: 50%;
  vertical-align: middle;
  padding: 2px;
  min-width: 20px;
  min-height: 20px;
  font-size: 12px;
  font-family: Arial, sans-serif;
  bottom: 100%;
  left: 100%;
  margin-left: -15px;
  margin-bottom: -15px;
  color: white;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
}

<div
  class="css-0"
>
  10
</div>
`;
